radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java

```diff
@@ -184,19 +212,59 @@ public void process(
     }
   }
 
+  private void persistCurrentSubstateAccumulatorHash(Transaction dbTxn) {
+    var key = new DatabaseEntry(SUBSTATE_ACCUMULATOR_HASH_KEY);
+    this.substateAccumulatorHashDatabase.put(
+        dbTxn, key, new DatabaseEntry(this.currentSubstateAccumulatorHash));
+  }
+
+  private void logEpochHash() {
+    logger.info(
+        "Epoch Hash: {} for epoch {}. Time spent since last epoch: {} s.",
+        Bytes.toHexString(this.currentSubstateAccumulatorHash),
+        lastEpochInDbOpt,
+        watch.elapsed().toSeconds());
+  }
+
+  private OperationStatus persistEpochHash(Transaction dbTxn) {
+    return epochHashDatabase.put(
+        dbTxn,
+        new DatabaseEntry(Longs.toByteArray(lastEpochInDbOpt.orElseThrow())),
+        new DatabaseEntry(this.currentSubstateAccumulatorHash));
+  }
+
+  private void handleEpochHashAccumulatorFileUpdate() {
+    // we do not overwrite hashes and only append in a sequential way
+    if (lastEpochInFileOpt.orElse(0L) == lastEpochInDbOpt.orElse(0L) - 1L) {
+      try {
+        writer.write(
```

Comment:
Can we flush the writer to disk before continuing? Else we might corrupt things

Reply:
Could you elaborate more on this?

Reply:
Calling the Java's flush method doesn't avoid this issue, but it is not a big deal in this case as we agreed. I'll add the flush.